packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -66,66 +59,46 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             bytes memory
         )
     {
-        bool isEthSign = _signatureType == Lib_OVMCodec.EOASignatureType.ETH_SIGNED_MESSAGE;
+        // Attempt to decode the transaction.
+        Lib_EIP155Tx.EIP155Tx memory transaction = Lib_EIP155Tx.decode(
+            _encodedTransaction,
+            Lib_ExecutionManagerWrapper.ovmCHAINID()
+        );
 
         // Address of this contract within the ovm (ovmADDRESS) should be the same as the
         // recovered address of the user who signed this message. This is how we manage to shim
         // account abstraction even though the user isn't a contract.
-        // Need to make sure that the transaction nonce is right and bump it if so.
         require(
-            Lib_ECDSAUtils.recover(
-                _transaction,
-                isEthSign,
-                _v,
-                _r,
-                _s
-            ) == address(this),
+            transaction.sender() == address(this),
             "Signature provided for EOA transaction execution is invalid."
         );
 
-        Lib_OVMCodec.EIP155Transaction memory decodedTx = Lib_OVMCodec.decodeEIP155Transaction(
-            _transaction,
-            isEthSign
-        );
-
-        // Grab the chain ID of the current network.
-        uint256 chainId;
-        assembly {
-            chainId := chainid()
-        }
-
-        // Need to make sure that the transaction chainId is correct.
-        require(
-            decodedTx.chainId == chainId,
-            "Transaction chainId does not match expected OVM chainId."
-        );
-
         // Need to make sure that the transaction nonce is right.
         require(
-            decodedTx.nonce == Lib_ExecutionManagerWrapper.ovmGETNONCE(),
+            transaction.nonce == Lib_ExecutionManagerWrapper.ovmGETNONCE(),
             "Transaction nonce does not match the expected nonce."
         );
 
         // TEMPORARY: Disable gas checks for mainnet.
         // // Need to make sure that the gas is sufficient to execute the transaction.
         // require(
-        //    gasleft() >= SafeMath.add(decodedTx.gasLimit, EXECUTION_VALIDATION_GAS_OVERHEAD),
+        //    gasleft() >= SafeMath.add(transaction.gasLimit, EXECUTION_VALIDATION_GAS_OVERHEAD),
         //    "Gas is not sufficient to execute the transaction."
         // );
 
         // Transfer fee to relayer.
         require(
             ovmETH.transfer(
                 msg.sender,
-                SafeMath.mul(decodedTx.gasLimit, decodedTx.gasPrice)
+                SafeMath.mul(transaction.gasLimit, transaction.gasPrice)
             ),
             "Fee was not transferred to relayer."
         );
 
         // Contract creations are signalled by sending a transaction to the zero address.
-        if (decodedTx.to == address(0)) {
+        if (transaction.to == address(0)) {
```

Comment:
Shouldn't this be the `isCreate` bool in `Lib_EIP155Tx.EIP155Tx`?

Reply:
Ah whoops, forgot to port that over. Thank you!

Reply:
Fixed